<launch>

  <!--Launch Turtlebot-->
  <include file="$(find turtlebot_bringup)/launch/minimal.launch"/>

  <!--Launch Realsense Camera-->
  <!--<include file="$(find realsense2_camera)/launch/rs_rgbd.launch"/>-->
  <node pkg="cv_camera" name="cv_camera" type="cv_camera_node" output="screen">
    <param name="device_id" value="3"/>
  </node>

  <!--Joy node for xbox 360-->
  <node pkg="joy" type="joy_node" name="joystick"/>

  <!--Custom Joy commander-->
  <node pkg="deep_turtle" type="custom_joy_cmd.py" name="joy_command" output="screen">
    <param name="joy_velocity_max" value="1.0"/>
    <param name="joy_omega_max" value="1.5"/>
    <param name="constant_vel" value="0.3"/>
    <remap from="turtle_cmd" to="/mobile_base/commands/velocity"/>
  </node>

  <!--Data Recorder-->
  <node pkg="deep_turtle" type="record_data_node.py" name="data_recorder" output="screen">
    <param name="test_name" value="wide_angle_2"/>
    <!--Record Every Nth frame-->
    <param name="record_n_frames" value="3"/>
    <remap from="record_data" to="/record_data"/>
    <remap from="rgb_image" to="/cv_camera/image_raw"/>
    <!--<remap from="depth_image" to="/camera/depth/image_rect_raw"/>-->
    <remap from="turtlebot_command" to="/mobile_base/commands/velocity"/>
  </node>

</launch>
